staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -242,9 +241,27 @@ func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision
 	if len(history) <= 0 {
 		return "No rollout history found.", nil
 	}
-	revisions := make([]int64, 0, len(history))
-	for _, revision := range history {
-		revisions = append(revisions, revision.Revision)
+
+	historyInfo := make(map[int64][]byte)
+	for _, h := range history {
+		historyInfo[h.Revision] = h.Data.Raw
+	}
+	if revision != 0 {
+		value, ok := historyInfo[revision]
+		if !ok {
+			return "", fmt.Errorf("unable to find the specified revision")
+		}
+		sts := appsv1.StatefulSet{}
+		err := json.Unmarshal(value, &sts)
```

Comment:
You should never use `json.Unmarshal` directly with any of the k8s resources. Have a look at how ds history is handling this inside `applyDaeomSetHistory` method.

Reply:
Thanks @soltysh. Created a method `applyStatefulSetHistory` (similar to `applyDaemonSetHistory`) and made the changes and amended to the same commit. Please have a look and provide your inputs.